compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -164,6 +164,16 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 ExprKind::Range(ref e1, ref e2, lims) => {
                     self.lower_expr_range(e.span, e1.as_deref(), e2.as_deref(), lims)
                 }
+                ExprKind::Underscore => {
+                    self.sess
+                        .struct_span_err(
+                            e.span,
+                            "expected expression, found reserved identifier `_`",
```

Comment:
This is not a good wording because it looks like a syntactic error.
Something like "`_` can only be used in assignment expressions" would be better.

Reply:
Reworded. Hopefully it is better now.